src/Microsoft.VisualStudio.AppDesigner/PropPages/PropertyControlData.vb

```diff
@@ -1537,9 +1538,24 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
             Debug.Assert(Extender IsNot Nothing)
 
             'Note: For property get we must pass in the extended object as the component.
-            Return Descriptor.GetValue(Extender)
+            Return GetValue(Descriptor, Extender)
```

Comment:
If the property is marked in the rule as Hidden the descriptor returns back an IEnumPage object which we can't decipher since we're expecting ints. That being said, there's still another problem. If the rule isn't hidden it's returning the string value and not the enum, I'm looking into that now.

For your other question, it shouldn't the code is a no-op if we get an actual value out of the descriptor.